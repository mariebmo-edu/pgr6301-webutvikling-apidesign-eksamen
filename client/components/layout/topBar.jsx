import { Link } from "react-router-dom";

export function TopBar({ user }) {
  const provider = user.microsoft
    ? "microsoft"
    : user.google
    ? "google"
    : "not logged in";

  return (
    <div className={"top-bar-container primary-background"}>
      <div className={"top-bar-content-container"}>
        <Link to={"/"} className={"not-link-link"}>
          <h1>IMPORTANT NEWS PAGE</h1>
        </Link>
        <div className={"logged-in-container"}>
          <Link to={"/"} className={"nav-button"}>
            Back
          </Link>
          {!user || Object.keys(user).length === 0 ? (
            <Link to={"/login"} className={"nav-button"}>
              Log in
            </Link>
          ) : (
            <div className={"nav-bar-title"}>
              <Link to={"/profile"} className={"nav-button"}>
                {user[provider].name}
              </Link>
              <Link
                to={"/login/endsession"}
                id={"log-out-btn"}
                className={"nav-button"}
              >
                Log out
              </Link>
              {user.microsoft && (
                <Link to={"/new"} className={"nav-button"}>
                  New Article
                </Link>
              )}
            </div>
          )}
        </div>
      </div>
    </div>
  );
}
