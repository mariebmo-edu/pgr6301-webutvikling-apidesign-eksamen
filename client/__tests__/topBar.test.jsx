import ReactDOM from "react-dom";
import { TopBar } from "../components/layout/topBar";
import { BrowserRouter as Router } from "react-router-dom";

const user = [
  {
    google: {
      picture: "/img/img",
      name: "test user",
      email: "test@user.com",
      authority: "User",
    },
  },

  {
    microsoft: {
      picture: "something",
      name: "test user 2",
      email: "test@user.com",
      authority: "Admin",
    },
  },
  {},
];

describe("top bar component", () => {
  it("handles admins", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <Router>
        <TopBar user={user[1]} />
      </Router>,
      domElement
    );

    expect(domElement).toMatchSnapshot();
    expect(
      Array.from(domElement.querySelectorAll(".nav-button")).map(
        (e) => e.innerHTML
      )
    ).toEqual(["Back", "test user 2", "Log out", "New Article"]);
  });

  it("handles users", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <Router>
        <TopBar user={user[0]} />
      </Router>,
      domElement
    );

    expect(domElement).toMatchSnapshot();
    expect(
      Array.from(domElement.querySelectorAll(".nav-button")).map(
        (e) => e.innerHTML
      )
    ).toEqual(["Back", "test user", "Log out"]);
  });

  it("handles others", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(
      <Router>
        <TopBar user={user[2]} />
      </Router>,
      domElement
    );

    expect(domElement).toMatchSnapshot();
    expect(
      Array.from(domElement.querySelectorAll(".nav-button")).map(
        (e) => e.innerHTML
      )
    ).toEqual(["Back", "Log in"]);
  });
});
