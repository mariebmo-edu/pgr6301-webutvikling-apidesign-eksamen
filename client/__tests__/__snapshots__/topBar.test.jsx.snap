// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`top bar component handles admins 1`] = `
<div>
  <div
    class="top-bar-container primary-background"
  >
    <div
      class="top-bar-content-container"
    >
      <a
        class="not-link-link"
        href="/"
      >
        <h1>
          IMPORTANT NEWS PAGE
        </h1>
      </a>
      <div
        class="logged-in-container"
      >
        <a
          class="nav-button"
          href="/"
        >
          Back
        </a>
        <div
          class="nav-bar-title"
        >
          <a
            class="nav-button"
            href="/profile"
          >
            test user 2
          </a>
          <a
            class="nav-button"
            href="/login/endsession"
            id="log-out-btn"
          >
            Log out
          </a>
          <a
            class="nav-button"
            href="/new"
          >
            New Article
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`top bar component handles others 1`] = `
<div>
  <div
    class="top-bar-container primary-background"
  >
    <div
      class="top-bar-content-container"
    >
      <a
        class="not-link-link"
        href="/"
      >
        <h1>
          IMPORTANT NEWS PAGE
        </h1>
      </a>
      <div
        class="logged-in-container"
      >
        <a
          class="nav-button"
          href="/"
        >
          Back
        </a>
        <a
          class="nav-button"
          href="/login"
        >
          Log in
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`top bar component handles users 1`] = `
<div>
  <div
    class="top-bar-container primary-background"
  >
    <div
      class="top-bar-content-container"
    >
      <a
        class="not-link-link"
        href="/"
      >
        <h1>
          IMPORTANT NEWS PAGE
        </h1>
      </a>
      <div
        class="logged-in-container"
      >
        <a
          class="nav-button"
          href="/"
        >
          Back
        </a>
        <div
          class="nav-bar-title"
        >
          <a
            class="nav-button"
            href="/profile"
          >
            test user
          </a>
          <a
            class="nav-button"
            href="/login/endsession"
            id="log-out-btn"
          >
            Log out
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
